File: program/find-gcd-of-two-numbers/find_gcd_of_two_numbers.rs
// Rust program to find the GCD 

use std::io;

fn main() 
{
    let mut n1:i32 = 0;
    let mut n2:i32 = 0;
    let mut rem:i32= 0;
    let mut x:i32  = 0;
    let mut y:i32  = 0;
    
    let mut input1 = String::new();
    let mut input2 = String::new();
    
    println!("Enter Number1: ");
    io::stdin().read_line(&mut input1).expect("Not a valid string");
    n1 = input1.trim().parse().expect("Not a valid number");

    println!("Enter Number2: ");
    io::stdin().read_line(&mut input2).expect("Not a valid string");
    n2 = input2.trim().parse().expect("Not a valid number");
    
    if (n1 > n2) 
    {
        x = n1;
        y = n2;
    }
    else 
    {
        x = n2;
        y = n1;
    }

    rem = x % y;

    while (rem != 0) {
        x = y;
        y = rem;
        rem = x % y;
    }

    println!("Greatest Common Divisor is: {}", y);
}

File: program/find-gcd-of-two-numbers/findGcdOfTwoNumbers.scala
object HelloWorld {
  
  def gcd(a:Int,b:Int):Int={
    if(b==0){
      return a
    }
    else{
      gcd(b,a%b)
    }
  }
  
	 def main(args: Array[String]): Unit=
	 {
	   var a = scala.io.StdIn.readInt
	   var b = scala.io.StdIn.readInt
	   
	    if (a == 0){
	      println( b)
	    }
	    
      if (b == 0){
        println( a)
      }
        
      println(gcd(a,b))
	 }
}
